app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/address/isUk.scala.html

```diff
@@ -43,12 +44,17 @@
     case _: AddressJourneyType.Returns.ChangingRepresenteeContactAddressJourney => ".representee"
 
     case c: AddressJourneyType.Returns.EnteringCompanyDetails =>
-      if(c.draftReturn.isLeft)                        ".multipleIndirect."
-      else if(c.representativeType.exists(_.isLeft))  ".companyDetails.personalRep."
-      else if(c.representativeType.exists(_.isRight)) ".companyDetails.capacitor."
-      else if(c.isATrust)                             ".companyDetails.trust."
-      else if(request.userType.exists(_ === Agent))   ".companyDetails.agent."
-      else                                            ".companyDetails."
+        if(c.draftReturn.isLeft)                       ".multipleIndirect."
+        else
+            c.representativeType match {
+                case Some(PersonalRepresentative) => ".companyDetails.personalRep."
+                case Some(PersonalRepresentativeInPeriodOfAdmin) => ".companyDetails.personalRep."
+                case Some(Capacitor) => ".companyDetails.capacitor."
+                case None =>
+                    if (c.isATrust) ".companyDetails.trust."
+                    else if (request.userType.exists(_ === Agent)) ".companyDetails.agent."
```

Comment:
As we are doing this logic twice, is it good to extract this into a variable
@isAgent = @{ request.userType.contains(UserType.Agent) }

Reply:
yeah, that's a good idea. I've factored it out now